contracts/pact_state_handlers_methods.go

```diff
@@ -0,0 +1,170 @@
+//
+// Copyright 2023 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package contracts
+
+import (
+	"context"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+
+	gomega "github.com/onsi/gomega"
+	models "github.com/pact-foundation/pact-go/v2/models"
+	appstudiov1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+var (
+	k8sClient client.Client
+	testEnv   *envtest.Environment
+	ctx       context.Context
+	cancel    context.CancelFunc
+)
+
+const timeout = 10 * time.Second
+const interval = 250 * time.Millisecond
+
+func createApp(setup bool, state models.ProviderState) (models.ProviderStateResponse, error) {
+	if !setup {
+		err := os.Setenv("SETUP", "false")
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		return nil, nil
+	}
+	err := os.Setenv("SETUP", "true")
+	gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+	params := parseApplication(state.Parameters)
+	hasApp := getApplicationSpec(params.appName, params.namespace)
+	hasAppLookupKey := types.NamespacedName{Name: params.appName, Namespace: params.namespace}
+	createdHasApp := &appstudiov1alpha1.Application{}
+
+	// create app
+	gomega.Expect(k8sClient.Create(ctx, hasApp)).Should(gomega.Succeed())
+
+	// check it is created
+	gomega.Eventually(func() bool {
+		err := k8sClient.Get(context.Background(), hasAppLookupKey, createdHasApp)
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
```

Comment:
You are not printing the error in gomega expect. In case of debug 0 information give this block